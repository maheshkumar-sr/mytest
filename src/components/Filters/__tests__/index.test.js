/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable quotes */
import React from "react";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import POCategoryPage, { isValid } from ".."; 

jest.mock(
  "../../../NikeCustomComponents/FormGroups/NikeSelectBox",
  () =>
    function SelectBox() {
      return <span>SelectBox</span>;
    }
);

jest.mock(
  "../../../NikeCustomComponents/DatePicker",
  () =>
    function DatePicker() {
      return <span>DatePicker</span>;
    }
);

jest.mock(
  "../../../NikeCustomComponents/FormGroups/NikeTextField",
  () =>
    function TextField() {
      return <span>TextField</span>;
    }
);

describe("Test isValid", () => {
  test("isValid function", () => {
    expect(
      isValid({
        poCategory: "none",
        shipTo: "none",
        material: "",
        season: "none",
        period: "none",
        initialStateValueFactCode: { factoryCode: "none" }
      })
    ).toBe(true);

    expect(
      isValid({
        poCategory: "none",
        shipTo: "none",
        material: "none",
        season: "",
        period: "none",
        initialStateValueFactCode: { factoryCode: "none" }
      })
    ).toBe(true);

    expect(
      isValid({
        poCategory: "none",
        shipTo: "none",
        material: "none",
        season: "none",
        period: "",
        initialStateValueFactCode: { factoryCode: "none" }
      })
    ).toBe(true);

    expect(
      isValid({
        poCategory: "none",
        shipTo: "none",
        material: "none",
        season: "none",
        period: "none",
        initialStateValueFactCode: { factoryCode: "none" }
      })
    ).toBe(true);
  });
}); 

describe("<POCategoryPage /> ", () => {
  let PROPS = {};
  beforeEach(() => {
    PROPS = {
      checked: true,
      handleChange: jest.fn(),
      name: "name",
      value: 1234,
      label: "label",
      disabled: true,
      subLabel: "subLabel",
      labelPlacement: "labelPlacement",
      isBlackFill: true,
      id: "PO1234",
      searchedResultSet: [
        { value: "P.O Creation", key: "podcsDate" },
        { value: "OGAC Date", key: "ogacDate" },
        { value: "GAC Date", key: "gacDate" }
      ],
      initialSearchedResultSet: [
        {
          season: 'Falls(FA)',
          gacDate: '2019-07-14T15:00:00.000Z',
          unitPrice: 254,
          ogacDate: '2019-07-14T15:00:00.000Z',
          trackingNo: 6948841345,
          poNumber: 6754887737,
          vendorName: 'LS',
          poCategory: 'Accepted',
          podcsDate: '2019-07-14T15:00:00.000Z',
          shipTo: 'NIKE JAPAN GROUP LLC',
          totalQuantity: 48,
          plantCode: 1084,
          material: 'AT5582-901'
        }
      ],
      onChangeFilters: jest.fn(),
      PoFilterOnChange: jest.fn()
    };
  }); 

  test("render the component correctly", () => {
    const wrapper = mount(<POCategoryPage {...PROPS} />);

    expect(wrapper.exists()).toBe(true);
  });

  test("handleChangeFactoryCode with null params", () => {
    PROPS.searchedPOCategory = "none";
    const wrapper = mount(<POCategoryPage {...PROPS} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test("render the component correctly", () => {
    const closeDrawerSpy = jest.fn();
    const wrapper = mount(
      <POCategoryPage {...PROPS} closeDrawer={closeDrawerSpy} />
    );

    wrapper.find("#filterButton").simulate("click");
    wrapper.find("#clearAllFilter").simulate("click");
    wrapper
      .find("#selectedFromDate")
      .props()
      .handleDateChange();

    wrapper
      .find("#selectedToDate")
      .props()
      .handleDateChange();

    wrapper
      .find("#factorycode")
      .props()
      .onChange({ preventDefault: jest.fn(), target: { value: "123456" } });

    wrapper
      .find("#factorycode")
      .props()
      .onChange({ preventDefault: jest.fn(), target: { value: "qnb123456" } });

    wrapper
      .find("#seasonListDropDown")
      .props()
      .handleChangeDropDown({ target: { value: "ereret" } });

    wrapper
      .find("#materialsDropDown")
      .props()
      .handleChangeDropDown({ target: { value: "ereret" } });

    wrapper
      .find("#shipToListDropDown")
      .props()
      .handleChangeDropDown({ target: { value: "ereret" } });

    wrapper
      .find("#periodListDropDown")
      .props()
      .handleChangeDropDown({ target: { value: "ereret" } });

    wrapper
      .find("#poCategoryListDropDown")
      .props()
      .handleChangeDropDown({ target: { value: "ereret" } }); 

    expect(wrapper.exists()).toBe(true);
  });
});