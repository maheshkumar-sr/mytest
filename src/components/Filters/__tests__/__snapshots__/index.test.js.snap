// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<POCategoryPage />  handleChangeFactoryCode with null params 1`] = `
<POCategoryPage
  PoFilterOnChange={[MockFunction]}
  checked={true}
  disabled={true}
  handleChange={[MockFunction]}
  id="PO1234"
  initialSearchedResultSet={
    Array [
      Object {
        "gacDate": "2019-07-14T15:00:00.000Z",
        "material": "AT5582-901",
        "ogacDate": "2019-07-14T15:00:00.000Z",
        "plantCode": 1084,
        "poCategory": "Accepted",
        "poNumber": 6754887737,
        "podcsDate": "2019-07-14T15:00:00.000Z",
        "season": "Falls(FA)",
        "shipTo": "NIKE JAPAN GROUP LLC",
        "totalQuantity": 48,
        "trackingNo": 6948841345,
        "unitPrice": 254,
        "vendorName": "LS",
      },
    ]
  }
  isBlackFill={true}
  label="label"
  labelPlacement="labelPlacement"
  name="name"
  onChangeFilters={
    [MockFunction] {
      "calls": Array [
        Array [
          Array [
            Object {
              "gacDate": "2019-07-14T15:00:00.000Z",
              "material": "AT5582-901",
              "ogacDate": "2019-07-14T15:00:00.000Z",
              "plantCode": 1084,
              "poCategory": "Accepted",
              "poNumber": 6754887737,
              "podcsDate": "2019-07-14T15:00:00.000Z",
              "season": "Falls(FA)",
              "shipTo": "NIKE JAPAN GROUP LLC",
              "totalQuantity": 48,
              "trackingNo": 6948841345,
              "unitPrice": 254,
              "vendorName": "LS",
            },
          ],
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  searchedPOCategory="none"
  searchedResultSet={
    Array [
      Object {
        "key": "podcsDate",
        "value": "P.O Creation",
      },
      Object {
        "key": "ogacDate",
        "value": "OGAC Date",
      },
      Object {
        "key": "gacDate",
        "value": "GAC Date",
      },
    ]
  }
  subLabel="subLabel"
  value={1234}
>
  <div
    className="ncss-container dropDown"
  >
    <div
      className="ncss-row dropDownPadding"
    >
      <div
        className="ncss-col-sm-1 itemDropDown"
      >
        <p
          className="filterName"
        >
          P.O Category
        </p>
        <SelectBox
          className="rootFilter"
          filterContainer={true}
          handleChangeDropDown={[Function]}
          id="poCategoryListDropDown"
          isSelectNative={false}
          searchType="none"
          selectFieldData={
            Array [
              Object {
                "key": "none",
                "value": "None",
              },
              Object {
                "key": undefined,
                "value": undefined,
              },
            ]
          }
          selectPlaceHolder="All P.Os"
        >
          <span>
            SelectBox
          </span>
        </SelectBox>
      </div>
      <div
        className="ncss-col-sm-1 itemDropDown"
      >
        <p
          className="filterName"
        >
          Period
        </p>
        <SelectBox
          className="rootFilter"
          filterContainer={true}
          handleChangeDropDown={[Function]}
          id="periodListDropDown"
          isSelectNative={false}
          searchType=""
          selectFieldData={
            Array [
              Object {
                "key": "podcsDate",
                "value": "P.O Creation",
              },
              Object {
                "key": "ogacDate",
                "value": "OGAC Date",
              },
              Object {
                "key": "gacDate",
                "value": "GAC Date",
              },
            ]
          }
          selectPlaceHolder="All P.Os"
        >
          <span>
            SelectBox
          </span>
        </SelectBox>
      </div>
      <div
        className="ncss-col-sm-1 itemDropDown"
      >
        <p
          className="filterName"
        >
          From
        </p>
        <DatePicker
          handleDateChange={[Function]}
          id="selectedFromDate"
          selectedDate={2020-04-02T08:07:06.618Z}
        >
          <span>
            DatePicker
          </span>
        </DatePicker>
      </div>
      <div
        className="ncss-col-sm-1 itemDropDown"
      >
        <p
          className="filterName"
        >
          To
        </p>
        <DatePicker
          handleDateChange={[Function]}
          id="selectedToDate"
          selectedDate={2020-04-02T08:07:06.618Z}
        >
          <span>
            DatePicker
          </span>
        </DatePicker>
      </div>
      <div
        className="ncss-col-sm-1 itemDropDown"
      >
        <p
          className="filterName"
        >
          Ship To
        </p>
        <SelectBox
          className="rootFilter"
          filterContainer={true}
          handleChangeDropDown={[Function]}
          id="shipToListDropDown"
          searchType=""
          selectFieldData={
            Array [
              Object {
                "key": "none",
                "value": "None",
              },
              Object {
                "key": undefined,
                "value": undefined,
              },
            ]
          }
          selectPlaceHolder="All P.Os"
        >
          <span>
            SelectBox
          </span>
        </SelectBox>
      </div>
      <div
        className="ncss-col-sm-1 itemDropDown"
      >
        <p
          className="filterName"
        >
          Material
        </p>
        <SelectBox
          className="rootFilter"
          filterContainer={true}
          handleChangeDropDown={[Function]}
          id="materialsDropDown"
          searchType=""
          selectFieldData={
            Array [
              Object {
                "key": "none",
                "value": "None",
              },
              Object {
                "key": undefined,
                "value": undefined,
              },
            ]
          }
          selectPlaceHolder="All P.Os"
        >
          <span>
            SelectBox
          </span>
        </SelectBox>
      </div>
      <div
        className="ncss-col-sm-1 itemDropDown"
      >
        <p
          className="filterName"
        >
          Season
        </p>
        <SelectBox
          className="rootFilter"
          filterContainer={true}
          handleChangeDropDown={[Function]}
          id="seasonListDropDown"
          searchType=""
          selectFieldData={
            Array [
              Object {
                "key": "none",
                "value": "None",
              },
              Object {
                "key": undefined,
                "value": undefined,
              },
            ]
          }
          selectPlaceHolder="All P.Os"
        >
          <span>
            SelectBox
          </span>
        </SelectBox>
      </div>
      <div
        className="ncss-col-sm-1 itemDropDown"
      >
        <p
          className="filterName"
        >
          Factory Code
        </p>
        <TextField
          error={false}
          helperText={null}
          id="factorycode"
          maxLength="4"
          name="factorycode"
          onChange={[Function]}
          type="text"
          value=""
        >
          <span>
            TextField
          </span>
        </TextField>
      </div>
    </div>
  </div>
  <button
    className="showAllFilters"
    id="filterButton"
    onClick={[Function]}
    type="button"
  >
    Show All Filters
  </button>
  <span
    className="divider"
  >
    |
  </span>
  <button
    className="clearAll"
    id="clearAllFilter"
    onClick={[Function]}
    type="button"
  >
    Clear all
  </button>
</POCategoryPage>
`;
